be2-scala/src/main/scala/ch/epfl/pop/model/network/method/message/data/DataBuilder.scala

```diff
@@ -0,0 +1,42 @@
+package ch.epfl.pop.model.network.method.message.data
+
+import ch.epfl.pop.model.network.method.message.data.lao.{CreateLao, StateLao, UpdateLao}
+import ch.epfl.pop.model.network.method.message.data.meeting.{CreateMeeting, StateMeeting}
+import ch.epfl.pop.model.network.method.message.data.rollCall.{CloseRollCall, CreateRollCall, OpenRollCall, ReopenRollCall}
+import ch.epfl.pop.model.network.method.message.data.witness.WitnessMessage
+
+object DataBuilder {
+  def buildData(messageData: MessageData, payload: String): Any = messageData._object match {
+    case ObjectType.LAO => buildLaoData(messageData, payload)
+    case ObjectType.MEETING => buildMeetingData(messageData, payload)
+    case ObjectType.ROLL_CALL => buildRollCallData(messageData, payload)
+    case ObjectType.MESSAGE => buildWitnessData(messageData, payload)
+    case _ => throw new ProtocolException(s"Unknown object '${messageData._object}' encountered while creating a Data")
+  }
+
+  def buildLaoData(messageData: MessageData, payload: String): Any = messageData.action match {
+    case ActionType.CREATE => CreateLao.buildFromJson(messageData, payload)
+    case ActionType.STATE => StateLao.buildFromJson(messageData, payload)
+    case ActionType.UPDATE_PROPERTIES => UpdateLao.buildFromJson(messageData, payload)
+    case _ => throw new ProtocolException(s"Unknown action '${messageData.action}' encountered while creating a Lao Data")
+  }
+
+  def buildMeetingData(messageData: MessageData, payload: String): Any = messageData.action match {
+    case ActionType.CREATE => CreateMeeting.buildFromJson(messageData, payload)
+    case ActionType.STATE => StateMeeting.buildFromJson(messageData, payload)
+    case _ => throw new ProtocolException(s"Unknown action '${messageData.action}' encountered while creating a Meeting Data")
+  }
+
+  def buildRollCallData(messageData: MessageData, payload: String): Any = messageData.action match {
+    case ActionType.CREATE => CreateRollCall.buildFromJson(messageData, payload)
+    case ActionType.OPEN => OpenRollCall.buildFromJson(messageData, payload)
+    case ActionType.REOPEN => ReopenRollCall.buildFromJson(messageData, payload)
```

Comment:
This doesn't match the new protocol, having only create/open/close messages for a roll call. Do we update that in this PR?

Reply:
@celinecamacho For now, I will not change it (because I don't know how exactly) so that this base PR can be merged. However, I will keep that in mind :) Thank you!